scripts/customEventHooks.lua

```diff
@@ -2,64 +2,194 @@ local customEventHooks = {}
 
 customEventHooks.validators = {}
 customEventHooks.handlers = {}
+customEventHooks.scriptID = {
+	handlers = {},
+	validators = {},
+	generatedScriptIDs = {}
+}
+
+function customEventHooks.generateScriptID(filePath)
+	if not string.match(filePath, "%S") then 
+		return nil
```

Comment:
should it be nil, or throw an error?

What is the likely hood this fails and why would this check fail?

Reply:
it should never fail, its just more added as a catch incase someone somehow access it directly and feeds  feeds it nothing, its sanitized elsewhere with our code

Reply:
I think we SHOULD throw then.

If someone is doing something in a bad way, we should try to fail early before we dive down a rabbit hole we shouldn't let them dig